src/environment.c

```diff
@@ -0,0 +1,345 @@
+    "\xb1\x19\x97\xfa\xff\x42\x0a\x33\x1b\xb4\xa4\xff\xdc\x8b\xdc\x8b\xa7\xc0"
+    "\x17\x32\xa9\x9a\x30\xd8\x3d\xbb\xeb\xd4\x69\x66\x6c\x84\xb4\x7d\x09\xd3"
+    "\xf5\xf4\x72\xb3\xb9\x38\x4a\xc6\x34\xbe\xba\x2a\x44\x0b\xa3\x6e\xc7\x66"
+    "\x11\x44\x13\x2f\x35\xe2\x06\x87\x35\x64";
+
+static const char SEED_ENV_NAME[] = "SPECULOS_SEED";
+
+static struct {
+  size_t size;
+  uint8_t seed[MAX_SEED_SIZE];
+} actual_seed = { 0 };
+
+/* APP NAME and VERSION */
+static const char APP_NAME_VERSION_ENV_NAME[] = "SPECULOS_APPNAME";
+static const char APP_NAME_VERSION_ENV_NAME_BKP[] = "SPECULOS_DETECTED_APPNAME";
+
+static env_sized_name_t app_name = { 3, "app\0" };
+static env_sized_name_t app_version = { 6, "1.33.7\0" };
+
+/* RNG VARIABLES */
+
+static const char RNG_ENV_NAME[] = "RNG_SEED";
+static unsigned int actual_rng = 0;
+
+/* ENDORSEMENT VARIABLES */
+
+static const char USER_KEY_ENV_NAME[] = "USER_PRIVATE_KEY";
+static const char ATTESTATION_ENV_NAME[] = "ATTESTATION_PRIVATE_KEY";
+
+static const uint8_t default_attestation_key[32] = {
+  0x13, 0x8f, 0xb9, 0xb9, 0x1d, 0xa7, 0x45, 0xf1, 0x29, 0x77, 0xa2,
+  0xb4, 0x6f, 0x0b, 0xce, 0x2f, 0x04, 0x18, 0xb5, 0x0f, 0xcb, 0x76,
+  0x63, 0x1b, 0xaf, 0x0f, 0x08, 0xce, 0xef, 0xdb, 0x5d, 0x57
+};
+
+static const uint8_t default_user_private_key[32] = {
+  0xe1, 0x5e, 0x01, 0xd4, 0x70, 0x82, 0xf0, 0xea, 0x47, 0x71, 0xc9,
+  0x9f, 0xe3, 0x12, 0xf9, 0xd7, 0x00, 0x93, 0xc8, 0x9a, 0xf4, 0x77,
+  0x87, 0xfd, 0xf8, 0x2e, 0x03, 0x1f, 0x67, 0x28, 0xb7, 0x10
+};
+
+static cx_ecfp_private_key_t attestation_key = { CX_CURVE_256K1, 32, {} };
+static cx_ecfp_private_key_t user_private_key_1 = { CX_CURVE_256K1, 32, {} };
+static cx_ecfp_private_key_t user_private_key_2 = { CX_CURVE_256K1, 32, {} };
+static env_user_certificate_t user_certificate_1 = { 0 };
+static env_user_certificate_t user_certificate_2 = { 0 };
+
+/* UTILS */
+
+static int unhex(uint8_t *dst, size_t dst_size, const char *src,
+                 size_t src_size)
+{
+  unsigned int i;
+  uint8_t acc;
+  int8_t c;
+
+  acc = 0;
+  for (i = 0; i < src_size && (i >> 1) < dst_size; i++) {
+    c = src[i];
+    switch (c) {
+    case '0' ... '9':
+      acc = (acc << 4) + c - '0';
+      break;
+    case 'a' ... 'f':
+      acc = (acc << 4) + c - 'a' + 10;
+      break;
+    case 'A' ... 'F':
+      acc = (acc << 4) + c - 'A' + 10;
+      break;
+    default:
+      return -1;
+    }
+
+    if (i % 2 != 0) {
+      dst[i >> 1] = acc;
+      acc = 0;
+    }
+  }
+
+  if (i != src_size) {
+    return -1;
+  }
+
+  return src_size / 2;
+}
+
+static void env_init_seed()
+{
+  ssize_t size;
+  char *p;
+
+  p = getenv(SEED_ENV_NAME);
+  if (p != NULL) {
+    size = unhex(actual_seed.seed, sizeof(actual_seed.seed), p, strlen(p));
+    if (size < 0) {
+      warnx("invalid seed passed through %s environment variable",
+            SEED_ENV_NAME);
+      p = NULL;
+    } else {
+      fprintf(stderr, "[*] Seed initialized from environment: '0x%s'\n", p);
```

Comment:
Maybe we should not show the seed value here?
In case someone use a real seed and copy paste the logs...

Reply:
> In case someone use a real seed and copy paste the logs...
:see_no_evil: 

Right, I guess beside testing purposes it's not very useful.